src/shogun/lib/SGMatrix.cpp

```diff
@@ -100,6 +100,19 @@ SGMatrix<T>::SGMatrix(const SGMatrix &orig) : SGReferencedData(orig)
 	copy_data(orig);
 }
 
+template <class T>
+SGMatrix<T>::SGMatrix(SGMatrix&& orig) noexcept
+	: SGReferencedData(std::move(orig)), matrix(orig.matrix),
+	  num_rows(orig.num_rows), num_cols(orig.num_cols),
+	  gpu_ptr(std::move(orig.gpu_ptr))
+{
+    orig.matrix = nullptr;
+    orig.num_rows = 0;
+    orig.num_cols = 0;
+	m_on_gpu.store(orig.m_on_gpu.load(
+			std::memory_order_acquire), std::memory_order_release);
+}
```

Comment:
move constructor for sgmatrix works fine. I am just not sure what happens when using GPU backend... In theory this is how it should be done!